app/controllers/application_controller.rb

```diff
@@ -91,6 +91,15 @@ def set_current_location_name
     cookies['location'] = params[:location] unless params[:location] == nil
   end
 
+  def check_for_empty_password
+    if session[:original_user] and session[:"warden.user.user.key"]
```

Comment:
I believe when I was testing a problem came up where after you did quick sign up, original_user was set but the user wasn't logged in because they hadn't confirmed. I'll do some testing on this and see if the warden part is actually needed.